lib/service/storage/elasticsearch.js

```diff
@@ -2294,13 +2387,27 @@ class ElasticSearch extends Service {
     for (let i = 0; i < documents.length; i++) {
       const document = documents[i];
 
-      if (!isPlainObject(document.body)) {
+      if (!isPlainObject(document.body) && !prepareMUpsert) {
         rejected.push({
           document,
           reason: 'document body must be an object',
           status: 400
         });
       }
+      else if (prepareMUpsert && !isPlainObject(document.changes)) {
```

Comment:
Nit: Best to maintain conditional consistency with previous condition (L2390).
```suggestion
      else if (!isPlainObject(document.changes) && prepareMUpsert) {
```